src/components/academy/game/backend/game-state.js

```diff
@@ -1,72 +1,89 @@
 import { storyXMLPathTest, storyXMLPathLive } from '../constants/constants'
 import { isStudent } from './user';
 
+var SaveManager = require('../save-manager/save-manager.js');
+
+
 /**
  * Handles data regarding the game state. 
  * - The student's list of completed quests and collectibles
  * - The student's current story mission
  * - The global list of missions that are open
+ * - The action to save user state to server.
  */
 let fetched = false;
-let studentMissionPointer = undefined,
-    studentData = undefined;
-export function fetchGameData(userStory, callback) {
+let studentData = undefined,
+    handleSaveData = undefined,
+    studentStory = undefined;
+
+export function fetchGameData(userStory, gameState, callback) {
   // fetch only needs to be called once; if there are additional calls somehow then ignore them
   if(fetched) {
     callback();
     return;
   }
   fetched = true;
-  studentMissionPointer = userStory.story;
-  // not implemented yet
-  studentData = undefined; // userStory.data;
+  studentStory = userStory.story;
+  studentData = gameState;
   fetchGlobalMissionPointer(callback);
 }
 
 // overrides
 let studentDataOverride = undefined,
-    missionPointerOverride = undefined,
-    currentDateOverride = undefined;
+    currentDateOverride = undefined,
+    studentStoryOverride = undefined;
+
 // override student game data
 export function overrideGameState(data) {
   if (data) {
-    studentDataOverride = data;
-    missionPointerOverride = data.missionPointer;
+    studentDataOverride = data.gameState;
+    studentStoryOverride = data.story;
     currentDateOverride = data.currentDate;
   } else {
-    studentDataOverride = missionPointerOverride = currentDateOverride = undefined;
+    studentStoryOverride = studentDataOverride = missionPointerOverride = currentDateOverride = undefined;
```

Comment:
Delete `missionPointerOverride` since we no longer use that name

Reply:
k done

Reply:
it's still here